From 2ce14d30f53ac6bbc128c3e1ee827b64c0f3322f Mon Sep 17 00:00:00 2001
From: Jiri Benc <jbenc@redhat.com>
Date: Fri, 24 Oct 2014 20:55:31 -0400
Subject: [net] udp_offload: Use IS_ERR_OR_NULL

Message-id: <72cccc35fc62bae1240c5620a16a7762f646261b.1414183890.git.jbenc@redhat.com>
Patchwork-id: 98461
O-Subject: [RHEL7.1 net PATCH 2/5] net/udp_offload: Use IS_ERR_OR_NULL
Bugzilla: 1144571
RH-Acked-by: Daniel Borkmann <dborkman@redhat.com>
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Jesper Brouer <jbrouer@redhat.com>
RH-Acked-by: Hannes Frederic Sowa <hannes@redhat.com>
RH-Acked-by: Jiri Pirko <jpirko@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1144571
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=8159555

commit 27446442a810f29d0fa97356bbc11f45e7ecfa6e
Author: Himangi Saraogi <himangi774@gmail.com>
Date:   Sun Jul 27 12:38:38 2014 +0530

    net/udp_offload: Use IS_ERR_OR_NULL

    This patch introduces the use of the macro IS_ERR_OR_NULL in place of
    tests for NULL and IS_ERR.

    The following Coccinelle semantic patch was used for making the change:

    @@
    expression e;
    @@

    - e == NULL || IS_ERR(e)
    + IS_ERR_OR_NULL(e)
     || ...

    Signed-off-by: Himangi Saraogi <himangi774@gmail.com>
    Acked-by: Julia Lawall <julia.lawall@lip6.fr>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/ipv4/udp.c b/net/ipv4/udp.c
index ed8053a..718970c 100644
--- a/net/ipv4/udp.c
+++ b/net/ipv4/udp.c
@@ -2371,7 +2371,7 @@ struct sk_buff *skb_udp_tunnel_segment(struct sk_buff *skb,
  /* segment inner packet. */
  enc_features = skb->dev->hw_enc_features & netif_skb_features(skb);
  segs = skb_mac_gso_segment(skb, enc_features);
- if (!segs || IS_ERR(segs)) {
+ if (IS_ERR_OR_NULL(segs)) {
   skb_gso_error_unwind(skb, protocol, tnl_hlen, mac_offset,
          mac_len);
   goto out;
-- 
1.7.1